sycl/source/detail/queue_impl.cpp

```diff
@@ -217,6 +217,11 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
     std::lock_guard<mutex_class> Lock(MMutex);
     Events = std::move(MEventsWeak);
     USMEvents = std::move(MEventsShared);
```

Comment:
Let's use `std::vector::swap` or `Events = vector_class<std::weak_ptr<event_impl>>(std::move(MEventsWeak));`
In move c'tor moved from std::vector is guaranteed to by empty.

Reply:
Using `std::vector::swap` sounds good.